workflows_core/dataset/field.py

```diff
@@ -216,14 +217,24 @@ def get_keyphrase(self, keyphrase_id: str):
             keyphrase_id=keyphrase_id,
         )
 
-    def update_keyphrase(self, keyphrase_id: str, update: dict):
-        return self._dataset.api._update_keyphrase(
-            dataset_id=self.dataset_id,
-            field=self._keyphrase_text_field,
-            alias=self._keyphrase_alias,
-            keyphrase_id=keyphrase_id,
-            update=update,
-        )
+    def update_keyphrase(self, keyphrase_id: str, update: Union[Keyphrase, dict]):
+        if isinstance(update, Keyphrase):
+            return self._dataset.api._update_keyphrase(
+                dataset_id=self.dataset_id,
+                field=self._keyphrase_text_field,
+                alias=self._keyphrase_alias,
+                keyphrase_id=keyphrase_id,
+                update=asdict(update),
```

Comment:
Looking at this code - it is clear that there is only 1 line difference. In order to ensure this is maintainable - you want to minimize the lines of code here. 
This if statement should only be 2 lines of code.

```
if isinstance(update, keyphrase):
      keyphrase = asdict(keyphrase)
```